### PR TITLE
Prepare wildfly codebase for new improved parser coming to core

### PR DESCRIPTION
**Pull request by ctomc**
Prepare full for wildfly/wildfly-core/pull/908

Mostly just changes to our testing templates to make sure we don't use default values.

extra commit https://github.com/wildfly/wildfly/commit/55d4f9e2c6a98883e8b0580a35a3e16ebe769f6a temporary disables testing properties attribute in IIOP subsystem as new parser differently (properly) orders writing of PropertiesAttributes and that causes failures in our subsystems tests.

Once Wildfly core with said PR is integrated to full, the second commit can and should be reverted.
https://issues.jboss.org/browse/WFLY-4987 is there to revert this commit.

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/InVMTransportDefinition.java [hunk 1]
```diff
@@ -42,7 +42,6 @@
 public class InVMTransportDefinition extends AbstractTransportDefinition {
 
     public static final SimpleAttributeDefinition SERVER_ID = create("server-id", INT)
-            .setDefaultValue(new ModelNode(0))
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/InVMTransportDefinition.java]: this is correct, this attribute must not have a default value :+1:

**Comment by ctomc** [comment]
failures are not related, there are two PRs open to @ignore them.

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java [hunk 2]
```diff
@@ -72,12 +72,12 @@
 
     protected static final AttributeDefinition SOCKET_BINDING = new SimpleAttributeDefinitionBuilder(
             Constants.ORB_SOCKET_BINDING, ModelType.STRING, true).setAttributeGroup(Constants.ORB)
-            .setDefaultValue(new ModelNode().set("iiop")).setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
```
Comment (kabir) [inline iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java]: This is nillable so it seems the default value should be left in place?

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java [hunk 3]
```diff
@@ -72,12 +72,12 @@
...
+            .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
             .addAccessConstraint(SensitiveTargetAccessConstraintDefinition.SOCKET_BINDING_REF).build();
 
     protected static final AttributeDefinition SSL_SOCKET_BINDING = new SimpleAttributeDefinitionBuilder(
             Constants.ORB_SSL_SOCKET_BINDING, ModelType.STRING, true).setAttributeGroup(Constants.ORB)
-            .setDefaultValue(new ModelNode().set("iiop-ssl")).setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
```
Comment (kabir) [inline iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java]: Nillable

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java (hunk 2 above)
Comment (kabir) [inline iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java]: If the intent is to make it required, we need a version bump since this has been in 9

File: iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd [hunk 4]
```diff
@@ -69,8 +70,9 @@
             </xs:documentation>
         </xs:annotation>
         <xs:attribute name="giop-version" type="xs:string" use="optional" default="1.2"/>
-        <xs:attribute name="socket-binding" type="xs:string" use="optional" default="iiop"/>
-        <xs:attribute name="ssl-socket-binding" type="xs:string" use="optional" default="iiop-ssl"/>
+        <xs:attribute name="socket-binding" type="xs:string" use="required" />
```
Comment (kabir) [inline iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd]: This and other changes need a new schema version

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml [hunk 5]
```diff
@@ -9,9 +9,8 @@
         <property name="some_property" value="some_value"/>
     </properties>
     <ior-settings>
-        <transport-config integrity="required" confidentiality="required" detect-replay="supported" detect-misordering="supported"
-                          trust-in-client="none" trust-in-target="none"/>
-        <as-context auth-method="username_password" realm="test_realm" required="true"/>
+        <transport-config integrity="required" confidentiality="required" detect-replay="supported" detect-misordering="supported" />
```
Comment (kabir) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml]: Why removing attributes?

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml [hunk 6]
```diff
@@ -9,9 +9,8 @@
...
+        <as-context realm="test_realm" required="true"/>
```
Comment (kabir) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml]: Removed attributes

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml [hunk 7]
```diff
@@ -9,9 +9,8 @@
         <property name="some_property" value="some_value"/>
     </properties>
     <ior-settings>
-        <transport-config integrity="required" confidentiality="required" detect-replay="supported" detect-misordering="supported"
-                          trust-in-client="none" trust-in-target="none"/>
-        <as-context auth-method="username_password" realm="test_realm" required="true"/>
+        <transport-config integrity="required" confidentiality="required" detect-replay="supported" detect-misordering="supported" />
```
Comment (kabir) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml]: Removed attributes?

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml [hunk 8]
```diff
@@ -9,9 +9,8 @@
...
+        <as-context realm="test_realm" required="true"/>
```
Comment (kabir) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml]: Removed attributes?

File: undertow/src/test/resources/org/wildfly/extension/undertow/undertow-3.0.xml [hunk 9]
```diff
@@ -86,7 +86,7 @@
                         http-only="true"
                         max-age="1000"/>
         <!--<persistent-sessions relative-to="${server.data.dir}" path="web-sessions"/>-->
-        <websockets dispatch-to-worker="true" buffer-pool="default" worker="default" />
```
Comment (kabir) [inline undertow/src/test/resources/org/wildfly/extension/undertow/undertow-3.0.xml]: Removed attributes?

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml (hunk 5 above)
Comment (ctomc) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml]: as they are redundant. They are all default values they mean noting if they are present.

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml (hunk 6 above)
Comment (ctomc) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem-ior-settings.xml]: yes as values here are default values

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml (hunk 7 above)
Comment (ctomc) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml]: default values, as such make no sense to be in configuration

File: legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml (hunk 8 above)
Comment (ctomc) [inline legacy/jacorb/src/test/resources/org/jboss/as/jacorb/subsystem.xml]: this are default values

File: iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd (hunk 4 above)
Comment (ctomc) [inline iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd]: i am quite indifferent about this, but having default values for socket binding has bitten us before. I don't mind reverting this change.

File: undertow/src/test/resources/org/wildfly/extension/undertow/undertow-3.0.xml (hunk 9 above)
Comment (ctomc) [inline undertow/src/test/resources/org/wildfly/extension/undertow/undertow-3.0.xml]: default values again, no need to be in xml

File: iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd (hunk 4 above)
Comment (ctomc) [inline iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd]: I've reverted the socket binding config to be optional again with default values

File: iiop-openjdk/src/main/resources/subsystem-templates/iiop-openjdk.xml [hunk 10]
```diff
@@ -3,7 +3,6 @@
 <config>
    <extension-module>org.wildfly.iiop-openjdk</extension-module>
    <subsystem xmlns="urn:jboss:domain:iiop-openjdk:1.0">
-       <orb socket-binding="iiop" ssl-socket-binding="iiop-ssl"/>
```
Comment (ctomc) [inline iiop-openjdk/src/main/resources/subsystem-templates/iiop-openjdk.xml]: this are default values, as such not needed to be in config by default.

Comment (kabir) [inline iiop-openjdk/src/main/resources/subsystem-templates/iiop-openjdk.xml]: But why remove them? That makes me suspicious :-)

File: iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd [hunk 11]
```diff
@@ -46,15 +46,16 @@
             </xs:documentation>
         </xs:annotation>
         <xs:sequence>
+            <xs:element name="properties" minOccurs="0" maxOccurs="1" type="genericPropertiesType"/>
```
Comment (kabir) [inline iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd]: If I am reading https://github.com/wildfly/wildfly/blob/master/iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPRootDefinition.java#L333 correctly, I think this one comes just before the transport-config element?

Comment (ctomc) [inline iiop-openjdk/src/main/resources/schema/jboss-as-iiop-openjdk_1_0.xsd]: Order is not controlled there.  at least not in this case.
When it comes to persisting attributes in context of a resource, order is as follows:
1) persist all attributes that don't belong to any attribute group and persist to xml attributes using the order attributes ware added to parser definition (not resource definition)
2) persist all attributes that don't belong to any attribute group and don't persist to xml attributes but to elements in order that ware added to parser definition
3) persist attribute groups (each group = xml element) and persist based on 1) & 2) 

so in short for this case, properties attribute doesn't belong to any attribute group so it is persisted directly after all other non grouped attributes on a resource.
In this case this is first child of the resource element.

File: iiop-openjdk/src/main/resources/subsystem-templates/iiop-openjdk.xml (hunk 10 above)
Comment (kabir) [inline iiop-openjdk/src/main/resources/subsystem-templates/iiop-openjdk.xml]: Sorry, that comment was in the wrong place. In the subsystem template that is fine, removing in tests I am not sure

Comment (ctomc) [inline iiop-openjdk/src/main/resources/subsystem-templates/iiop-openjdk.xml]: In main test tests everything I modified xml it to use non default values. 
In some other tests I removed the redundant parts as they didn't affect the test in any way.
